src/Transports.AspNetCore/GraphQLHttpMiddlewareOptions.cs

```diff
@@ -0,0 +1,107 @@
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+
+namespace GraphQL.Server.Transports.AspNetCore;
+
+/// <summary>
+/// Configuration options for <see cref="GraphQLHttpMiddleware"/>.
+/// </summary>
+public class GraphQLHttpMiddlewareOptions
+{
+    /// <summary>
+    /// Enables handling of GET requests.
+    /// </summary>
+    public bool HandleGet { get; set; } = true;
+
+    /// <summary>
+    /// Enables handling of POST requests, including form submissions, JSON-formatted requests and raw query requests.
+    /// </summary>
+    public bool HandlePost { get; set; } = true;
+
+    /********** WebSockets support ********
+    
+    /// <summary>
+    /// Enables handling of WebSockets requests.
+    /// <br/><br/>
+    /// Requires calling <see cref="WebSocketMiddlewareExtensions.UseWebSockets(IApplicationBuilder)"/>
+    /// to initialize the WebSocket pipeline within the ASP.NET Core framework.
+    /// </summary>
+    public bool HandleWebSockets { get; set; } = true;
+
+    ******************/
+
+    /// <summary>
+    /// Enables handling of batched GraphQL requests for POST requests when formatted as JSON.
+    /// </summary>
+    public bool EnableBatchedRequests { get; set; } = true;
+
+    /// <summary>
+    /// Enables parallel execution of batched GraphQL requests.
+    /// </summary>
+    public bool BatchedRequestsExecuteInParallel { get; set; } = true;
+
+    /// <summary>
+    /// When enabled, GraphQL requests with validation errors
+    /// have the HTTP status code set to 400 Bad Request.
+    /// GraphQL requests with execution errors are unaffected.
+    /// <br/><br/>
+    /// Does not apply to batched or WebSocket requests.
+    /// </summary>
+    public bool ValidationErrorsReturnBadRequest { get; set; } = true;
+
+    /// <summary>
+    /// Enables parsing the query string on POST requests.
+    /// If enabled, the query string properties override those in the body of the request.
+    /// </summary>
+    public bool ReadQueryStringOnPost { get; set; } = true;
+
+    /// <summary>
+    /// Enables reading variables from the query string.
+    /// Variables are interpreted as JSON and deserialized before being
+    /// provided to the <see cref="IDocumentExecuter"/>.
+    /// </summary>
+    public bool ReadVariablesFromQueryString { get; set; } = true;
+
+    /// <summary>
+    /// Enables reading extensions from the query string.
+    /// Extensions are interpreted as JSON and deserialized before being
+    /// provided to the <see cref="IDocumentExecuter"/>.
+    /// </summary>
+    public bool ReadExtensionsFromQueryString { get; set; } = true;
+
+    /// <summary>
+    /// If set, requires that <see cref="IIdentity.IsAuthenticated"/> return <see langword="true"/>
+    /// for the user within <see cref="HttpContext.User"/>
+    /// prior to executing the GraphQL request or accepting the WebSocket connection.
+    /// </summary>
+    public bool AuthorizationRequired { get; set; }
```

Comment:
*Authorization*Required but you talk about `IsAuthenticated` property in comment and call OnNot*Authenticated* in `AuthorizationParameters`. So... `AuthorizationRequired` -> `AuthenticationRequired` ?

Reply:
That is technically true, but:

1. Naming it as `AuthorizationRequired` aligns it with the other two properties, so intellisense will show all three properties as you start to type `Authorize`.

2. The attribute used to mark graphs as requiring authentication is `AuthorizeAttribute` (or `[Authorize]`) which is the same as in ASP.NET Core -- even though technically as you say we are not requiring authorization but merely authentication.

3. For the same reasons as above, the extension method to require authentication is called `graphType.Authorize()`.

I did think about this in the design process; I would not change it.

Reply:
> so intellisense will show all three properties as you start to type Authorize.

Usually I stop typing on _Auth_ ))

> 2 + 3

Agree. People using this code may have the same questions as mine so I suggest to append note in xml comments about the reasons for the selected naming.

Reply:
Whatever changes you suggest to the xml comment in this regard will be fine.